src/ccl_halomod.c

```diff
@@ -0,0 +1,428 @@
+#include "ccl_utils.h"
+#include "math.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "gsl/gsl_integration.h"
+#include "gsl/gsl_interp.h"
+#include "gsl/gsl_spline.h"
+#include "gsl/gsl_sf_expint.h"
+#include "ccl_placeholder.h"
+#include "ccl_background.h"
+#include "ccl_power.h"
+#include "ccl_massfunc.h"
+#include "ccl_error.h"
+#include "class.h"
+#include "ccl_params.h"
+#include "ccl_emu17.h"
+#include "ccl_emu17_params.h"
+
+// Set two-halo term
+// 1 - Standard two-halo term
+// 2 - Linear theory
+int i2h=1;
+
+// Set full halo model pwoer spectrum method
+// 1 - Standard sum of two- and one-halo terms
+// 2 - Smooth transition using alpha parameter from Mead et al. (2015)
+int ipow=1;
+
+// Set concentration-mass relation
+// 1 - Bhattaharya et al. (2011)
+// 2 - Full Bullock et al. (2001)
+// 3 - Duffy et al. (2008)
+// 4 - Constant concentration (for testing)
+// 5 - Simple Bullock et al. (2001)
+int iconc=3;
+
+// Select window function
+// 1 - NFW profile, appropriate for matter power spectrum
+int iwin=1;
+
+// Set the mass range for the halo-model integration
+double mmin=1e7;
+double mmax=1e17;
+
+// Virial density of haloes
+double Delta_v=200.;
+
+// HMcode parameter: one-halo damping wavenumber: k*  
+double k_star(ccl_cosmology *cosmo, double a, int * status){  
+  return 0.584/ccl_sigmaVz(cosmo, 0., a, status);
+}
```

Comment:
If a function is not gonna be used by users, it should be made static.
Also, it seems like you're not even using it, so get rid of anything you don't use.

Reply:
Removed things I don't use (into a new branch) and made other functions explicitly static.

Reply:
Done